src/components/common/ThreeToggle/index.tsx

```diff
@@ -0,0 +1,152 @@
+import { useTheme } from 'next-themes';
+import { useEffect, useId, useState } from 'react';
+import MonitorIcon from '../../../../public/assets/icons/monitor.svg';
+import MoonIcon from '../../../../public/assets/icons/moon.svg';
+import SunIcon from '../../../../public/assets/icons/sun.svg';
+import styles from './style.module.scss';
+
+const ThreeToggle = () => {
+  const { theme, setTheme, systemTheme } = useTheme();
+  const [mounted, setMounted] = useState(false);
+
+  const lightId = `light${useId()}`;
+  const systemId = `system${useId()}`;
+  const darkId = `light${useId()}`;
+
+  const iconSize = 30;
+
+  const calcIndicatorStyle = (newTheme: string | undefined) => {
+    switch (newTheme) {
+      case 'light':
+        return styles.switchOne;
+      case 'system':
+        return styles.switchTwo;
+      case 'dark':
+        return styles.switchThree;
+      default:
+        return styles.switchOne;
+    }
+  };
+
+  const calcIconStyle = (newTheme: string | undefined) => {
+    switch (newTheme) {
+      case 'light':
+        return styles.iconLightMode;
+      case 'system':
+        return systemTheme === 'light' ? styles.iconLightMode : styles.iconDarkMode;
+      case 'dark':
+        return styles.iconDarkMode;
+      default:
+        return styles.iconDarkMode;
+    }
+  };
+
+  // IF YOU WANT TO ADD BACKGROUND TO SWITCH, USE THIS FUNCTION
+  // const calcSwitchBackgroundStyle = (newTheme: string | undefined) => {
+  //   switch (newTheme) {
+  //     case 'light':
+  //       return styles.switchLightMode;
+  //     case 'system':
+  //       return systemTheme === 'light' ? styles.switchLightMode : styles.switchDarkMode;
+  //     case 'dark':
+  //       return styles.switchDarkMode;
+  //     default:
+  //       return styles.switchDarkMode;
+  //   }
+  // };
+
+  const calcIndColorStyle = (newTheme: string | undefined) => {
+    switch (newTheme) {
+      case 'light':
+        return styles.indicatorLightMode;
+      case 'system':
+        return systemTheme === 'light' ? styles.indicatorLightMode : styles.indicatorDarkMode;
+      case 'dark':
+        return styles.indicatorDarkMode;
+      default:
+        return styles.indicatorDarkMode;
+    }
+  };
+
+  const calcAltText = (theme: string | undefined) => {
+    switch (theme) {
+      case 'light':
+        return 'Icon representing light theme is on';
+      case 'system':
+        return 'Icon representing system theme is on';
+      case 'dark':
+        return 'Icon representing dark theme is on';
+      default:
+        return 'Icon representing dark theme is on';
```

Comment:
Out of curiosity, why does it default to dark theme? Is this just to satisfy TypeScript?

Reply:
This was mainly to satisfy TypeScript yes. It should never trigger though.